src/MVDegrains.h

```diff
@@ -65,32 +65,16 @@ static void Degrain_sse2(uint8_t *pDst, int nDstPitch, const uint8_t *pSrc, int
 
     __m128i zero = _mm_setzero_si128();
     __m128i wsrc = _mm_set1_epi16(WSrc);
-    __m128i wrefs[12];
-    wrefs[0] = _mm_set1_epi16(WRefs[0]);
-    wrefs[1] = _mm_set1_epi16(WRefs[1]);
-    //TODO: Test sticking this in a loop. If performance is the same, a loop will massively reduce the amount of code required.
-    if (radius > 1) {
-        wrefs[2] = _mm_set1_epi16(WRefs[2]);
-        wrefs[3] = _mm_set1_epi16(WRefs[3]);
-    }
-    if (radius > 2) {
-        wrefs[4] = _mm_set1_epi16(WRefs[4]);
-        wrefs[5] = _mm_set1_epi16(WRefs[5]);
-    }
-    if (radius > 3) {
-        wrefs[6] = _mm_set1_epi16(WRefs[6]);
-        wrefs[7] = _mm_set1_epi16(WRefs[7]);
-    }
-    if (radius > 4) {
-        wrefs[8] = _mm_set1_epi16(WRefs[8]);
-        wrefs[9] = _mm_set1_epi16(WRefs[9]);
-    }
-    if (radius > 5) {
-        wrefs[10] = _mm_set1_epi16(WRefs[10]);
-        wrefs[11] = _mm_set1_epi16(WRefs[11]);
+    __m128i wrefs[radius * 2];
+
+    // We intentionally jump by 2 (here and below), as it delineates groups of
+    // backward/forward and ALSO produces testably faster code.
+    for(int i = 0; i < radius * 2; i += 2) {
+        wrefs[i] = _mm_set1_epi16(WRefs[i]);
+        wrefs[i + 1] = _mm_set1_epi16(WRefs[i + 1]);
     }
 
-    __m128i src, refs[12];
+    __m128i src, accum, refs[radius * 2];
```

Comment:
Does this work with MSVC? I think dynamic arrays are not standard C++.

Reply:
Good question. I don’t have a windows dev environment setup offhand, so that’s something I can look into.

Reply:
According to stackoverflow, it's not going to work with Microsoft's compiler. It doesn't allow variable-length arrays.

Reply:
Reverted.